flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyPartitionRequestClient.java

```diff
@@ -277,6 +270,16 @@ private boolean canBeReused() {
         return clientFactory.isConnectionReuseEnabled() && !clientHandler.hasChannelError();
     }
 
+    @VisibleForTesting
+    public Channel getTcpChannel() {
+        return tcpChannel;
+    }
+
+    @VisibleForTesting
```

Comment:
These two `@VisibleForTesting` methods should probably be avoided. For the connection reuse scenario, we are doing black box testing and should rely less on internal state to make assertions. I suggest that we consider another approach to trigger the close of the serverHandler.